src/sentry/integrations/github_enterprise/webhook.py

```diff
@@ -0,0 +1,194 @@
+
+
+def get_installation_metadata(event):
+    try:
+        integration = Integration.objects.get(external_id=event['installation']['id'])
+    except Integration.DoesNotExist:
+        return
+    return integration.metadata['installation']
+
+
+class GitHubEnterpriseInstallationEventWebhook(InstallationEventWebhook):
+    provider = 'github-enterprise'
+    repo_provider = 'github_enterprise'
+
+
+class GitHubEnterpriseInstallationRepositoryEventWebhook(InstallationRepositoryEventWebhook):
+    provider = 'github-enterprise'
+    repo_provider = 'github_enterprise'
+
+    # https://developer.github.com/v3/activity/events/types/#installationrepositoriesevent
+    def _handle(self, event, organization, repo):
+        pass
+
+
+class GitHubEnterprisePushEventWebhook(PushEventWebhook):
+    provider = 'github-enterprise'
+    repo_provider = 'github_enterprise'
+
+    # https://developer.github.com/v3/activity/events/types/#pushevent
+    def is_anonymous_email(self, email):
+        return email[-25:] == '@users.noreply.github.com'
+
+    def get_external_id(self, username):
+        return 'github_enterprise:%s' % username
+
+    def get_client(self, event):
+        metadata = get_installation_metadata(event)
+        if metadata is None:
+            return HttpResponse(status=400)
+
+        return GitHubEnterpriseAppsClient(
+            metadata['url'],
+            metadata['id'],
+            event['installation']['id'],
+            metadata['private_key'])
+
+    def should_ignore_commit(self, commit):
+        return GitHubEnterpriseRepositoryProvider.should_ignore_commit(commit['message'])
+
+
+class GitHubEnterprisePullRequestEventWebhook(PullRequestEventWebhook):
+    provider = 'github-enterprise'
+    repo_provider = 'github_enterprise'
+
+    # https://developer.github.com/v3/activity/events/types/#pullrequestevent
+    def is_anonymous_email(self, email):
+        return email[-25:] == '@users.noreply.github.com'
+
+    def get_external_id(self, username):
+        return 'github_enterprise:%s' % username
+
+
+class GithubWebhookBase(View):
+    # https://developer.github.com/webhooks/
+    def get_handler(self, event_type):
+        return self._handlers.get(event_type)
+
+    def is_valid_signature(self, method, body, secret, signature):
+        if method == 'sha1':
+            mod = hashlib.sha1
+        else:
+            raise NotImplementedError('signature method %s is not supported' % (method, ))
+        expected = hmac.new(
+            key=secret.encode('utf-8'),
+            msg=body,
+            digestmod=mod,
+        ).hexdigest()
+        return constant_time_compare(expected, signature)
+
+    @method_decorator(csrf_exempt)
+    def dispatch(self, request, *args, **kwargs):
+        if request.method != 'POST':
+            return HttpResponse(status=405)
+
+        return super(GithubWebhookBase, self).dispatch(request, *args, **kwargs)
+
+    def get_logging_data(self):
```

Comment:
lol just realized this isn't implemented for gh either. we can probably get rid of this if it's not being used